datumaro/plugins/market1501_format.py

```diff
@@ -3,137 +3,148 @@
 
             item = items.get(item_id)
             if item is None:
-                item = DatasetItem(id=item_id, subset=self._subset,
-                    image=image_path)
+                item = DatasetItem(id=item_id, subset=subset, image=image_path,
+                    attributes=attributes)
                 items[item_id] = item
 
-            if pid != Market1501Path.UNKNOWN_ID or \
-                    camid != Market1501Path.UNKNOWN_ID:
-                attributes = item.attributes
-                attributes['query'] = subdir == Market1501Path.QUERY_DIR
-                attributes['person_id'] = pid
-                attributes['camera_id'] = camid
         return items
 
 class Market1501Importer(Importer):
     @classmethod
     def find_sources(cls, path):
-        if not osp.isdir(path):
-            return []
-        return [{ 'url': path, 'format': Market1501Extractor.NAME }]
+        for dirname in os.listdir(path):
+            if dirname.startswith((Market1501Path.BBOX_DIR,
+                    Market1501Path.QUERY_DIR, Market1501Path.LIST_PREFIX)):
+                return [{'url': path, 'format': Market1501Extractor.NAME}]
 
 class Market1501Converter(Converter):
     DEFAULT_IMAGE_EXT = Market1501Path.IMAGE_EXT
 
+    def _make_dir_name(self, item):
+        dirname = Market1501Path.BBOX_DIR + item.subset
+        query = item.attributes.get('query')
+        if query is not None and isinstance(query, str):
+            query = strtobool(query)
+        if query:
+            dirname = Market1501Path.QUERY_DIR
+        return dirname
+
     def apply(self):
         for subset_name, subset in self._extractor.subsets().items():
             annotation = ''
+            used_frames = {}
 
             for item in subset:
+                dirname = self._make_dir_name(item)
+
                 image_name = item.id
-                if Market1501Path.PATTERN.search(image_name) is None:
-                    if 'person_id' in item.attributes and \
-                            'camera_id' in item.attributes:
-                        image_pattern = '{:04d}_c{}s1_000000_00{}'
-                        pid = int(item.attributes['person_id'])
-                        camid = int(item.attributes['camera_id']) + 1
-                        dirname, basename = osp.split(item.id)
-                        image_name = osp.join(dirname,
-                            image_pattern.format(pid, camid, basename))
-
-                dirname = Market1501Path.BBOX_DIR + subset_name
-                if 'query' in item.attributes:
-                    query = item.attributes.get('query')
-                    if isinstance(query, str):
-                        query = strtobool(query)
-                    if query:
-                        dirname = Market1501Path.QUERY_DIR
+                pid = item.attributes.get('person_id')
+                match = Market1501Path.PATTERN.fullmatch(item.id)
+                if not match and pid is not None:
```

Comment:
```suggestion
                if not match and pid:
```
I'm not sure we can use empty `pid`